source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -31,6 +35,121 @@ def __init__(self, obj, position, _rangeObj=None):
 					1
 				)
 
+	def move(self, unit, direction, endPoint=None):
+		oldRange=None
+		if not self._isCaret:
+			# Insure we haven't gone beyond the visible text.
+			# UIA adds thousands of blank lines to the end of the console.
+			visiRanges = self.obj.UIATextPattern.GetVisibleRanges()
+			if visiRanges.length > 0:
```

Comment:
I don't know how caching works here, but it might make sense to do something like this:

```suggestion

			visibleRangesLength = visiRanges.Length
			if visibleRangesLength > 0:
```

Reply:
What exactly is the advantage of doing this? We only use the value once per function call.

Reply:
You're also getting the length for lastVisiRange